pkg/api/v1/duration.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+// Duration implements a JSON serializable time.Duration.
+type Duration time.Duration
```

Comment:
I believe Google secret manager uses the [protocol buffer duration package](https://pkg.go.dev/github.com/golang/protobuf/ptypes/duration), so I think it's a good idea to have something custom here to help us coordinate between the requests coming in from the client and the format expected on the backend.

Reply:
Good note! I'll add a method to translate this type into pb duration.